### ISSUE TITLE
[BUG] Sextant sample navigation crashes with Android.Util.AndroidRuntimeException: 'Animators may only be run on Looper threads'

### ISSUE THREAD

**Issue by winterdouglas** [reactiveui/Sextant#171]
**Describe the bug**
Navigating on the Sextant Android sample makes the app crash with the exception `Android.Util.AndroidRuntimeException: 'Animators may only be run on Looper threads'` when running in a `ReactiveCommand`.

**Steps To Reproduce**
1. Run the SextantSample.Android
2. On the home page, tap on any of the buttons to navigate
3. The app right away crashes with the exception `Android.Util.AndroidRuntimeException: 'Animators may only be run on Looper threads'`

**Expected behavior**
It shouldn't crash :)

**Environment**
- OS:       Android
- Device:   Tested on emulator, Huawei P20
- Version:  The ones that use RxUI >= 10.1.1
- Working Version:

**Comment by winterdouglas** [comment]
I did a backtracking investigation already as follows:
### Versions of libraries
- Xamarin.Forms 4.2.0.910310
- ReactiveUI.XamForms 10.5.43
    - System.Reactive 4.3.1
- Android Support 28.0.0.3
---
### Investigation step by step
- Navigating by using the `Navigation` from the page  itself works nicely
- Changing the `ReactiveCommand` with an `ICommand`: the issue doesn't happen anymore
- At this point I can conclude that the _navigation is not the issue_ but something related to RxUI, Rx.Net or Xamarin.Forms
- Next downgrading RxUI to the latest 9.x (9.22.1): **The issue to go away**
- Just to make sure it's not Xamarin.Forms I manually changed the version of Xamarin to the initial one mentioned above (4.2.0.910310): **The issue still persists**
- Downgrading RxUI to 10.5.31: **Same results**
- Downgrading RxUI down to the version 10.1.7: **Same results** _(checked on 10.5.7, 10.2.2, 10.1.7)_
- RxUI 10.1.5: **Gives me another exception**: `System.TypeInitializationException: 'The type initializer for 'ReactiveUI.RxApp' threw an exception.'`
- RxUI 10.1.1: The error above happens on this version too
- RxUI 10.0.1: Here it **works** again
- RxUI 9.22.1: **Works as well**
#### All the above steps were performed with clean and rebuilt project + the app being reinstalled (uninstall/redeploy)

Apparently the issue started from the version 10.1.1 on, the only change from the version 10.0.1 to 10.1.1 is the update of `Mono.Android` on the following commit:
https://github.com/reactiveui/ReactiveUI/commit/c4be8bcf5f65e217fb1419057c4d3d4cf1785554

**Comment by winterdouglas** [comment]
Relates to:
#reactiveui/ReactiveUI/issues/2170

**Comment by RLittlesII** [comment]
@winterdouglas Thanks for looking into this issue and the detailed report.

**Comment by winterdouglas** [comment]
I did another round of investigation here and so far I'd say the issue is really related to the command's `IsExecuting`.

```
var canExecute = Observable
    .Return(false)
    .Delay(TimeSpan.FromSeconds(5), RxApp.TaskpoolScheduler)
    .StartWith(true)
    .ObserveOn(RxApp.MainThreadScheduler);

PushPage = ReactiveCommand
    .CreateFromObservable(() =>
        Observable.Return(Unit.Default),
        canExecute,
        outputScheduler: RxApp.MainThreadScheduler);
```

I can reproduce the issue with this code above. As you can see the command does nothing and the only thing that happens there is the command's `canExecute` being  initialized as `true` and after 5 seconds going to `false`, that's enough to make the app to crash.
I could conclude also that the issue happens only when there's an UI element bound to the command.

**Comment by winterdouglas** [comment]
Going a bit further with the investigation I could find that by doing the following, the code actually breaks as well, so the real issue is not on the command's `IsExecuting` but on the `RxApp.MainThreadScheduler` that's actually not really the main thread.

```
Observable
    .Return(false)
    .Delay(TimeSpan.FromSeconds(5), RxApp.TaskpoolScheduler)
    .ObserveOn(RxApp.MainThreadScheduler)
    .Subscribe(x =>
    {
        var isMain = Xamarin.Essentials.MainThread.IsMainThread;
        PushPage.IsEnabled = x;
    });
```

I'm gonna create a sample with these cases and some unit tests as @RLittlesII suggested.

**Comment by winterdouglas** [comment]
Ok, I now created a completely clean project with only `ReactiveUI` and put the exact same package versions and the code as above, I couldn't reproduce the issue anymore!

Looking further what could be the difference among them I could now find the real issue, it happens because on the Sextant sample the target/compile framework for the Android project is Android 8.1 while in the newly created one is Android 9, that makes the difference.
I noticed that when the project targets **Android 8.1 a `DefaultScheduler`** is resolved while when targeting **Android 9 the correct scheduler is resolved**.

**Sextant sample (compiled using Android 8.1)**
![sextant-sample](https://user-images.githubusercontent.com/1329194/70391852-e9d6fd80-19d9-11ea-93cc-be16e9fb15a8.png)

**Clean sample (compiled using Android 9)**
![clean-sample](https://user-images.githubusercontent.com/1329194/70391863-02dfae80-19da-11ea-936f-460404dcf2ef.png)

**Comment by RLittlesII** [comment]
Resolved By: #172
